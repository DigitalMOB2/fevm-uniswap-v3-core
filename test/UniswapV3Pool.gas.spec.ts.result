

  UniswapV3Pool gas tests
    fee is off
      #swapExact0For1
        ✔ first swap in block with no tick movement (4210ms)
        ✔ first swap in block moves tick, no initialized crossings (4182ms)
        ✔ second swap in block with no tick movement (8359ms)
        ✔ second swap in block moves tick, no initialized crossings (4499ms)
        ✔ first swap in block, large swap, no initialized crossings (4146ms)
        ✔ first swap in block, large swap crossing several initialized ticks (4577ms)
        ✔ first swap in block, large swap crossing a single initialized tick (4266ms)
        ✔ second swap in block, large swap crossing several initialized ticks (4615ms)
        ✔ second swap in block, large swap crossing a single initialized tick (4415ms)
        ✔ large swap crossing several initialized ticks after some time passes (4608ms)
        ✔ large swap crossing several initialized ticks second time after some time passes (4860ms)
      #mint
        around current price
          ✔ new position mint first in range (4180ms)
          ✔ add to position existing (4365ms)
          ✔ second position in same range (4310ms)
          ✔ add to position after some time passes (4463ms)
        below current price
          ✔ new position mint first in range (4132ms)
          ✔ add to position existing (4264ms)
          ✔ second position in same range (4415ms)
          ✔ add to position after some time passes (4313ms)
        above current price
          ✔ new position mint first in range (4239ms)
          ✔ add to position existing (4276ms)
          ✔ second position in same range (4319ms)
          ✔ add to position after some time passes (4339ms)
      #burn
        around current price
          ✔ burn when only position using ticks (4116ms)
          ✔ partial position burn (4107ms)
          ✔ entire position burn but other positions are using the ticks (4259ms)
          ✔ burn entire position after some time passes (4145ms)
        below current price
          ✔ burn when only position using ticks (4080ms)
          ✔ partial position burn (4138ms)
          ✔ entire position burn but other positions are using the ticks (4355ms)
          ✔ burn entire position after some time passes (4147ms)
        above current price
          ✔ burn when only position using ticks (4102ms)
          ✔ partial position burn (4078ms)
          ✔ entire position burn but other positions are using the ticks (4260ms)
          ✔ burn entire position after some time passes (4143ms)
      #poke
        ✔ best case (4496ms)
      #collect
        ✔ close to worst case (4402ms)
      #increaseObservationCardinalityNext
        ✔ grow by 1 slot (4117ms)
        ✔ no op (4069ms)
      #snapshotCumulativesInside
        ✔ tick inside
        ✔ tick above (250ms)
        ✔ tick below (239ms)
    fee is on
      #swapExact0For1
        ✔ first swap in block with no tick movement (4124ms)
        ✔ first swap in block moves tick, no initialized crossings (4143ms)
        ✔ second swap in block with no tick movement (8323ms)
        ✔ second swap in block moves tick, no initialized crossings (4246ms)
        ✔ first swap in block, large swap, no initialized crossings (4138ms)
        ✔ first swap in block, large swap crossing several initialized ticks (4416ms)
        ✔ first swap in block, large swap crossing a single initialized tick (4240ms)
        ✔ second swap in block, large swap crossing several initialized ticks (4647ms)
        ✔ second swap in block, large swap crossing a single initialized tick (4417ms)
        ✔ large swap crossing several initialized ticks after some time passes (4571ms)
        ✔ large swap crossing several initialized ticks second time after some time passes (4805ms)
      #mint
        around current price
          ✔ new position mint first in range (4156ms)
          ✔ add to position existing (4314ms)
          ✔ second position in same range (4334ms)
          ✔ add to position after some time passes (4419ms)
        below current price
          ✔ new position mint first in range (4171ms)
          ✔ add to position existing (4257ms)
          ✔ second position in same range (4303ms)
          ✔ add to position after some time passes (4343ms)
        above current price
          ✔ new position mint first in range (4166ms)
          ✔ add to position existing (4272ms)
          ✔ second position in same range (4229ms)
          ✔ add to position after some time passes (4324ms)
      #burn
        around current price
          ✔ burn when only position using ticks (4113ms)
          ✔ partial position burn (4095ms)
          ✔ entire position burn but other positions are using the ticks (4260ms)
          ✔ burn entire position after some time passes (4191ms)
        below current price
          ✔ burn when only position using ticks (4087ms)
          ✔ partial position burn (4130ms)
          ✔ entire position burn but other positions are using the ticks (4244ms)
          ✔ burn entire position after some time passes (4151ms)
        above current price
          ✔ burn when only position using ticks (4132ms)
          ✔ partial position burn (4090ms)
          ✔ entire position burn but other positions are using the ticks (4200ms)
          ✔ burn entire position after some time passes (4143ms)
      #poke
        ✔ best case (4542ms)
      #collect
        ✔ close to worst case (4391ms)
      #increaseObservationCardinalityNext
        ✔ grow by 1 slot (4080ms)
        ✔ no op (4093ms)
      #snapshotCumulativesInside
        ✔ tick inside
        ✔ tick above (238ms)
        ✔ tick below (232ms)


Snapshot Summary
 › 30 snapshots updated from 1 test suite.

  84 passing (42m)

