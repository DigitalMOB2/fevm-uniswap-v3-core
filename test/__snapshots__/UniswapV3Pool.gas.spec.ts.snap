// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniswapV3Pool gas tests fee is off #burn above current price burn entire position after some time passes 1`] = `18672773`;

exports[`UniswapV3Pool gas tests fee is off #burn above current price burn when only position using ticks 1`] = `18672773`;

exports[`UniswapV3Pool gas tests fee is off #burn above current price entire position burn but other positions are using the ticks 1`] = `22332154`;

exports[`UniswapV3Pool gas tests fee is off #burn above current price partial position burn 1`] = `23337860`;

exports[`UniswapV3Pool gas tests fee is off #burn around current price burn entire position after some time passes 1`] = `21333647`;

exports[`UniswapV3Pool gas tests fee is off #burn around current price burn when only position using ticks 1`] = `19707782`;

exports[`UniswapV3Pool gas tests fee is off #burn around current price entire position burn but other positions are using the ticks 1`] = `23368876`;

exports[`UniswapV3Pool gas tests fee is off #burn around current price partial position burn 1`] = `24574154`;

exports[`UniswapV3Pool gas tests fee is off #burn below current price burn entire position after some time passes 1`] = `19850122`;

exports[`UniswapV3Pool gas tests fee is off #burn below current price burn when only position using ticks 1`] = `19850122`;

exports[`UniswapV3Pool gas tests fee is off #burn below current price entire position burn but other positions are using the ticks 1`] = `25164350`;

exports[`UniswapV3Pool gas tests fee is off #burn below current price partial position burn 1`] = `26193002`;

exports[`UniswapV3Pool gas tests fee is off #collect close to worst case 1`] = `26003938`;

exports[`UniswapV3Pool gas tests fee is off #increaseObservationCardinalityNext grow by 1 slot 1`] = `12393176`;

exports[`UniswapV3Pool gas tests fee is off #increaseObservationCardinalityNext no op 1`] = `12071471`;

exports[`UniswapV3Pool gas tests fee is off #mint above current price add to position after some time passes 1`] = `52116468`;

exports[`UniswapV3Pool gas tests fee is off #mint above current price add to position existing 1`] = `52116476`;

exports[`UniswapV3Pool gas tests fee is off #mint above current price new position mint first in range 1`] = `51211300`;

exports[`UniswapV3Pool gas tests fee is off #mint above current price second position in same range 1`] = `51065974`;

exports[`UniswapV3Pool gas tests fee is off #mint around current price add to position after some time passes 1`] = `71287156`;

exports[`UniswapV3Pool gas tests fee is off #mint around current price add to position existing 1`] = `69663679`;

exports[`UniswapV3Pool gas tests fee is off #mint around current price new position mint first in range 1`] = `72662639`;

exports[`UniswapV3Pool gas tests fee is off #mint around current price second position in same range 1`] = `71084236`;

exports[`UniswapV3Pool gas tests fee is off #mint below current price add to position after some time passes 1`] = `54765033`;

exports[`UniswapV3Pool gas tests fee is off #mint below current price add to position existing 1`] = `54765025`;

exports[`UniswapV3Pool gas tests fee is off #mint below current price new position mint first in range 1`] = `56677646`;

exports[`UniswapV3Pool gas tests fee is off #mint below current price second position in same range 1`] = `56165118`;

exports[`UniswapV3Pool gas tests fee is off #poke best case 1`] = `16187615`;

exports[`UniswapV3Pool gas tests fee is off #snapshotCumulativesInside tick above 1`] = `7989545`;

exports[`UniswapV3Pool gas tests fee is off #snapshotCumulativesInside tick below 1`] = `7973282`;

exports[`UniswapV3Pool gas tests fee is off #snapshotCumulativesInside tick inside 1`] = `8998713`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 first swap in block moves tick, no initialized crossings 1`] = `53483959`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 first swap in block with no tick movement 1`] = `50316071`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 first swap in block, large swap crossing a single initialized tick 1`] = `61597122`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 first swap in block, large swap crossing several initialized ticks 1`] = `79419521`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 first swap in block, large swap, no initialized crossings 1`] = `56262835`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 large swap crossing several initialized ticks after some time passes 1`] = `79416113`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 large swap crossing several initialized ticks second time after some time passes 1`] = `83477217`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 second swap in block moves tick, no initialized crossings 1`] = `53482111`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 second swap in block with no tick movement 1`] = `50395011`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 second swap in block, large swap crossing a single initialized tick 1`] = `58976744`;

exports[`UniswapV3Pool gas tests fee is off #swapExact0For1 second swap in block, large swap crossing several initialized ticks 1`] = `76780019`;

exports[`UniswapV3Pool gas tests fee is on #burn above current price burn entire position after some time passes 1`] = `19047578`;

exports[`UniswapV3Pool gas tests fee is on #burn above current price burn when only position using ticks 1`] = `19047578`;

exports[`UniswapV3Pool gas tests fee is on #burn above current price entire position burn but other positions are using the ticks 1`] = `22705043`;

exports[`UniswapV3Pool gas tests fee is on #burn above current price partial position burn 1`] = `23713962`;

exports[`UniswapV3Pool gas tests fee is on #burn around current price burn entire position after some time passes 1`] = `21697432`;

exports[`UniswapV3Pool gas tests fee is on #burn around current price burn when only position using ticks 1`] = `20056623`;

exports[`UniswapV3Pool gas tests fee is on #burn around current price entire position burn but other positions are using the ticks 1`] = `23775590`;

exports[`UniswapV3Pool gas tests fee is on #burn around current price partial position burn 1`] = `24956684`;

exports[`UniswapV3Pool gas tests fee is on #burn below current price burn entire position after some time passes 1`] = `20223043`;

exports[`UniswapV3Pool gas tests fee is on #burn below current price burn when only position using ticks 1`] = `20223043`;

exports[`UniswapV3Pool gas tests fee is on #burn below current price entire position burn but other positions are using the ticks 1`] = `25545776`;

exports[`UniswapV3Pool gas tests fee is on #burn below current price partial position burn 1`] = `26573044`;

exports[`UniswapV3Pool gas tests fee is on #collect close to worst case 1`] = `26189507`;

exports[`UniswapV3Pool gas tests fee is on #increaseObservationCardinalityNext grow by 1 slot 1`] = `12491605`;

exports[`UniswapV3Pool gas tests fee is on #increaseObservationCardinalityNext no op 1`] = `12169848`;

exports[`UniswapV3Pool gas tests fee is on #mint above current price add to position after some time passes 1`] = `52599653`;

exports[`UniswapV3Pool gas tests fee is on #mint above current price add to position existing 1`] = `52599653`;

exports[`UniswapV3Pool gas tests fee is on #mint above current price new position mint first in range 1`] = `51697085`;

exports[`UniswapV3Pool gas tests fee is on #mint above current price second position in same range 1`] = `51502371`;

exports[`UniswapV3Pool gas tests fee is on #mint around current price add to position after some time passes 1`] = `71808417`;

exports[`UniswapV3Pool gas tests fee is on #mint around current price add to position existing 1`] = `70184460`;

exports[`UniswapV3Pool gas tests fee is on #mint around current price new position mint first in range 1`] = `73128676`;

exports[`UniswapV3Pool gas tests fee is on #mint around current price second position in same range 1`] = `71609184`;

exports[`UniswapV3Pool gas tests fee is on #mint below current price add to position after some time passes 1`] = `55247718`;

exports[`UniswapV3Pool gas tests fee is on #mint below current price add to position existing 1`] = `56768952`;

exports[`UniswapV3Pool gas tests fee is on #mint below current price new position mint first in range 1`] = `57160723`;

exports[`UniswapV3Pool gas tests fee is on #mint below current price second position in same range 1`] = `55656354`;

exports[`UniswapV3Pool gas tests fee is on #poke best case 1`] = `16576828`;

exports[`UniswapV3Pool gas tests fee is on #snapshotCumulativesInside tick above 1`] = `8001641`;

exports[`UniswapV3Pool gas tests fee is on #snapshotCumulativesInside tick below 1`] = `7985822`;

exports[`UniswapV3Pool gas tests fee is on #snapshotCumulativesInside tick inside 1`] = `9003113`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 first swap in block moves tick, no initialized crossings 1`] = `57457430`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 first swap in block with no tick movement 1`] = `51776666`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 first swap in block, large swap crossing a single initialized tick 1`] = `63169833`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 first swap in block, large swap crossing several initialized ticks 1`] = `81109114`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 first swap in block, large swap, no initialized crossings 1`] = `57810521`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 large swap crossing several initialized ticks after some time passes 1`] = `81105722`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 large swap crossing several initialized ticks second time after some time passes 1`] = `85140653`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 second swap in block moves tick, no initialized crossings 1`] = `54938994`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 second swap in block with no tick movement 1`] = `51855622`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 second swap in block, large swap crossing a single initialized tick 1`] = `60531544`;

exports[`UniswapV3Pool gas tests fee is on #swapExact0For1 second swap in block, large swap crossing several initialized ticks 1`] = `78447872`;
